operator/src/main/java/org/projectnessie/operator/resource/NessieAuthentication.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.operator.resource;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import io.fabric8.generator.annotation.Default;
+import io.soabase.recordbuilder.core.RecordBuilder;
+
+@RecordBuilder
+@JsonInclude(Include.NON_NULL)
+public record NessieAuthentication(
```

Comment:
In case anybody wonders, the CRD generator is very stubborn and does not work with `@Values.Immutable` at all. It wants POJOs. Luckily enough, records do work very well.